src/components/Calendar.tsx

```diff
@@ -0,0 +1,150 @@
+import { View, Pressable } from "react-native";
+import Text from "@/components/Text";
+import { useState } from "react";
+import {
+  Directions,
+  Gesture,
+  GestureDetector,
+} from "react-native-gesture-handler";
+import { runOnJS } from "react-native-reanimated";
+
+const DAYS_IN_WEEK = 7;
+const MONTHS_IN_YEAR = 12;
+const SWIPE_THRESHOLD = 50;
+const MONTH_NAMES = [
+  "jan",
+  "feb",
+  "mar",
+  "apr",
+  "may",
+  "jun",
+  "jul",
+  "aug",
+  "sep",
+  "oct",
+  "nov",
+  "dec",
+];
+
+interface CalendarProps {
+  selectedDate: string;
+  onDateSelect: (date: string) => void;
+}
+
+export default function Calendar({
+  selectedDate,
+  onDateSelect,
+}: CalendarProps) {
+  const currentDate = new Date();
+  const initialDate =
+    selectedDate === ""
+      ? new Date(currentDate.getFullYear(), currentDate.getMonth())
+      : new Date(selectedDate);
+  const [date, setDate] = useState<Date>(initialDate);
+  const month = date.getMonth();
+  const year = date.getFullYear();
+
+  // returns a 2D array of days in DAYS_IN_WEEK columns
```

Comment:
Idea: there's a nice js libary called `date-fns`. You may be able to use some fns in there to clean up your code if you want too